[executorch] Delete core FunctionRef

swolchok opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #3452

It was previously replaced with a function pointer typedef; let's just use function pointers directly and have one less type to worry about.

Differential Revision: [D56846841](https://our.internmc.facebook.com/intern/diff/D56846841/)

swolchok committed:
[executorch] Delete core FunctionRef

It was previously replaced with a function pointer typedef; let's just use function pointers directly and have one less type to worry about.

Differential Revision: [D56846841](https://our.internmc.facebook.com/intern/diff/D56846841/)

[ghstack-poisoned]

larryliu0820 reviewed (approved):
As long as CI jobs are green let's do it

swolchok committed:
format on "[executorch] Delete core FunctionRef"


It was previously replaced with a function pointer typedef; let's just use function pointers directly and have one less type to worry about.

Differential Revision: [D56846841](https://our.internmc.facebook.com/intern/diff/D56846841/)

[ghstack-poisoned]

swolchok committed:
Update on "[executorch] Delete core FunctionRef"


It was previously replaced with a function pointer typedef; let's just use function pointers directly and have one less type to worry about.

Differential Revision: [D56846841](https://our.internmc.facebook.com/intern/diff/D56846841/)

[ghstack-poisoned]

swolchok committed:
fix pattern build on "[executorch] Delete core FunctionRef"


It was previously replaced with a function pointer typedef; let's just use function pointers directly and have one less type to worry about.

Differential Revision: [D56846841](https://our.internmc.facebook.com/intern/diff/D56846841/)

[ghstack-poisoned]

swolchok committed:
Update on "[executorch] Delete core FunctionRef"


It was previously replaced with a function pointer typedef; let's just use function pointers directly and have one less type to worry about.

Differential Revision: [D56846841](https://our.internmc.facebook.com/intern/diff/D56846841/)

[ghstack-poisoned]
